Makefile

```diff
@@ -120,22 +119,29 @@ fix_cformat:
 $(C_LIB_CARGO_PATH):
 	cargo rustc --release --crate-type cdylib -p qiskit-cext
 
-$(C_QISKIT_H): $(C_LIB_CARGO_PATH)
-	cbindgen --crate qiskit-cext --output $(C_DIR_INCLUDE)/qiskit.h --lang C
+# Compile the C library and move the header into the right location, if it was generated
+$(C_LIB_CARGO_PATH): $(C_DIR_INCLUDE)
+	cargo build --release -p qiskit-cext
+	@if [ -f crates/cext/qiskit.h ]; then mv crates/cext/qiskit.h $(C_DIR_INCLUDE)/qiskit.h; fi
 
 $(C_DIR_LIB):
 	mkdir -p $(C_DIR_LIB)
 
-$(C_LIBQISKIT): $(C_DIR_LIB) $(C_LIB_CARGO_PATH)
+$(C_DIR_INCLUDE):
+	mkdir -p $(C_DIR_INCLUDE)
+
+$(C_LIBQISKIT): $(C_DIR_LIB)  $(C_LIB_CARGO_PATH)
 	cp $(C_LIB_CARGO_PATH) $(C_DIR_LIB)/$(subst _cext,,$(C_LIB_CARGO_FILENAME))
 
-.PHONY: cheader clib c
+$(C_QISKIT_H): $(C_LIB_CARGO_PATH)
+	@if [ ! -f "$(C_QISKIT_H)" ]; then cbindgen --crate qiskit-cext --output $(C_DIR_INCLUDE)/qiskit.h; fi
+
+.PHONY: c cheader
 cheader: $(C_QISKIT_H)
-clib: $(C_LIBQISKIT)
-c: clib cheader
+c: $(C_LIBQISKIT) $(C_QISKIT_H)
 
 # Use ctest to run C API tests
-ctest: $(C_QISKIT_H)
+ctest: $(C_LIB_CARGO_PATH)
```

Comment:
If `ctest` does depend on `C_QISKIT_H` being installed into the `dist` directory, it should depend on that as well as the library being in the `dist` directory.